File: lista3/zadanie1/Kolo.java
public class Kolo extends Figura {

    private int promien;

    Kolo(int promien){
        if (promien < 0) throw new IllegalArgumentException(promien + "-> niedodatni promien :c");
        this.promien = promien;
    }

    public int dajPromien(){
        return promien;
    }

    public double obliczPole(){
        return (promien * promien * Math.PI);
    }

    public double obliczObwod(){
        return (2 * promien * Math.PI);
    }

    public void dajNazwe(){
        System.out.println("Jestem kolem");
    }
}

File: lista3/zadanie2/Kolo.cpp
#include "Figura.hpp"
#include "Kolo.hpp"
#include <cmath>

double Kolo::obliczPole(){
    if (promien <= 0) 
        throw std::invalid_argument(std::to_string(promien) + std::string("-> Otrzymano niedodatni promien"));
    return promien * promien * M_PI ;
}

double Kolo::obliczObwod(){
    if (promien <= 0) 
        throw std::invalid_argument(std::to_string(promien) + std::string("-> Otrzymano niedodatni promien"));
    return 2 * promien * M_PI;
}

void Kolo::dajNazwe(){
    std::cout << "Jestem kolem" << std::endl;
}